localstack/services/sns/provider.py

```diff
@@ -1559,3 +1588,81 @@ def unsubscribe_sqs_queue(queue_url):
             sub_url = subscriber.get("sqs_queue_url") or subscriber["Endpoint"]
             if queue_url == sub_url:
                 subscriptions.remove(subscriber)
+
+
+@singleton_factory
```

Comment:
i know i said to use the singleton factory in the last review :D but i think we can just remove it (sice we only call register_sns_api_resources once in `on_after_init`)